File: ProductsService2/src/main/java/com/appsdeveloperblog/estore/productsservice2/data/repo/ProductRepository.java
package com.appsdeveloperblog.estore.productsservice2.data.repo;

import com.appsdeveloperblog.estore.productsservice2.data.entity.ProductEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ProductRepository extends JpaRepository<ProductEntity, String> {

    ProductEntity findByProductId(String productId);

    ProductEntity findByProductIdOrTitle(String productId, String title);
}

File: ProductsService/src/main/kotlin/com/appsdeveloperblog/estore/productsservice/data/repo/ProductRepository.kt
package com.appsdeveloperblog.estore.productsservice.data.repo

import com.appsdeveloperblog.estore.productsservice.data.ProductEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ProductRepository : JpaRepository<ProductEntity, String> {
    fun findByProductId(productId: String): MutableList<ProductEntity>
    fun findByProductIdOrTitle(productId: String, title: String): ProductEntity?
}